In [2]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import seaborn 
import sqlite3
import plotly.express as px
from matplotlib import pyplot as plt
import matplotlib.pyplot as plt


# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

path = "../input/basketball/" #Everything preceding the file name
database = path + 'nba.sqlite' #The path + the file name

/kaggle/input/basketball/nba.sqlite
/kaggle/input/basketball/csv/other_stats.csv
/kaggle/input/basketball/csv/draft_history.csv
/kaggle/input/basketball/csv/team_details.csv
/kaggle/input/basketball/csv/player.csv
/kaggle/input/basketball/csv/draft_combine_stats.csv
/kaggle/input/basketball/csv/team_info_common.csv
/kaggle/input/basketball/csv/common_player_info.csv
/kaggle/input/basketball/csv/team_history.csv
/kaggle/input/basketball/csv/game.csv
/kaggle/input/basketball/csv/game_summary.csv
/kaggle/input/basketball/csv/officials.csv
/kaggle/input/basketball/csv/game_info.csv
/kaggle/input/basketball/csv/line_score.csv
/kaggle/input/basketball/csv/play_by_play.csv
/kaggle/input/basketball/csv/team.csv
/kaggle/input/basketball/csv/inactive_players.csv


# Overview
# These are the questions I will answer through this analysis.

**Decada 80**

Question 1: Which team won the most home games in the 1980s?

Question 2: Which team won the most away games in the 1980s?

**Decada 90**

Question 4: Which team won the most home games in the 1990s?

Question 5: Which team won the most away games in the 1990s?


**Decada 2000** 

Question 6: Which team won the most home games in the 2000s?

Question 7: Which team won the most away games in the 2000s?

**Decada 2010** 

Question 8: Which team won the most home games in the 2010s?

Question 9: Which team won the most away games in the 2010s?


# *Português (Brasil)*
Visão geral

Estas são as questões que irei responder através desta análise.

**Decada 80**

Questão 1: Qual time ganhou mais jogos em casa na década 80?

Questão 2: Qual time que mais venceu jogos em fora de casa na década de 80? 

**Decada 90**

Questão 3: Qual time ganhou mais jogos em casa na década 90?

Questão 4: Qual time que mais venceu jogos em fora de casa na década de 90? 


**Decada 2000**

Questão 5: Qual time ganhou mais jogos em casa na década 2000?

Questão 6: Qual time que mais venceu jogos em fora de casa na década de 2000?


**Decada 2010**

Questão 7: Qual time ganhou mais jogos em casa na década 2010?

Questão 8: Qual time que mais venceu jogos em fora de casa na década de 2010?



# Conecta a base sqlite3

In [3]:
### connect to database, select all from player table - print players

conn = sqlite3.connect(database)



# **Decada 80 - Question 1**

Which team won the most home games in the 1980s

Qual time ganhou mais jogos em casa na década de 80?

In [ ]:
games80_win_home = pd.read_sql(
    """SELECT resultado.team_abbreviation_home, resultado.team_name_home, max(resultado.TOTAL_WIN_HOME) as total_win_home
        FROM
        (SELECT team_abbreviation_home, team_name_home, COUNT(*) AS TOTAL_WIN_HOME
      FROM GAME 
       WHERE GAME.game_date BETWEEN "1980-01-01" AND "1989-12-31"
        AND GAME.wl_home = "W"
 GROUP BY GAME.team_abbreviation_home, GAME.team_id_home) as resultado; """, conn)


print(games80_win_home)

**Boston Celtics  - 315 home win**

**Boston Celtics - 315 vitórias em casa**

# **Decada 80 - Question 2** 

Which team won the most away games in the 1980s?

Qual time ganhou mais jogos fora de casa na década de 80?

In [ ]:
players = pd.read_sql("""SELECT * FROM player;   """, conn)
games80_win_away = pd.read_sql(
    """SELECT resultado.team_abbreviation_away, resultado.team_name_away, max(resultado.TOTAL_WIN_AWAY) as total_win_away
        FROM
        (SELECT team_abbreviation_away, team_name_away, COUNT(*) AS TOTAL_WIN_AWAY
      FROM GAME 
       WHERE GAME.game_date BETWEEN "1980-01-01" AND "1989-12-31"
        AND GAME.wl_away = "W"
 GROUP BY GAME.team_abbreviation_away, GAME.team_id_away) as resultado; """, conn)

print(games80_win_away)

**Los Angeles Lakers  - 232 away win**

**Los Angeles Lakers - 232 vitórias fora de casa**

# **Decada 90 - Question 3**

Which team won the most home games in the 1990s?

Qual time ganhou mais jogos em casa na década de 90?

In [ ]:
games90_win_home = pd.read_sql(
    """SELECT resultado.team_abbreviation_home, resultado.team_name_home, max(resultado.TOTAL_WIN_HOME) as total_win_home
        FROM
        (SELECT team_abbreviation_home, team_name_home, COUNT(*) AS TOTAL_WIN_HOME
      FROM GAME 
       WHERE GAME.game_date BETWEEN "1990-01-01" AND "1999-12-31"
        AND GAME.wl_home = "W"
 GROUP BY GAME.team_abbreviation_home, GAME.team_id_home) as resultado; """, conn)


print(games90_win_home)

**Chicago Bulls - 310 home win**

**Chicago Bulls - 315 vitórias em casa**

# **Decada 90 - Question 4**


Which team won the most away games in the 1990s?

Qual time ganhou mais jogos fora de casa na década de 90?

In [ ]:
# **Decada 90 - Question 4**

Which team won the most away games in the 1990s?

Qual time ganhou mais jogos fora de casa na década de 90?

In [ ]:

games90_win_away = pd.read_sql(
    """SELECT resultado.team_abbreviation_away, resultado.team_name_away, max(resultado.TOTAL_WIN_AWAY) as total_win_away
        FROM
        (SELECT team_abbreviation_away, team_name_away, COUNT(*) AS TOTAL_WIN_AWAY
      FROM GAME 
       WHERE GAME.game_date BETWEEN "1990-01-01" AND "1999-12-31"
        AND GAME.wl_away = "W"
 GROUP BY GAME.team_abbreviation_away, GAME.team_id_away) as resultado; """, conn)

print(games90_win_away)

**Chicago Bulls - 231 away win**

**Chicago Bulls - 213 vitórias fora de casa**

# **Decada 2000 - Question 5**

Which team won the most home games in the 2000s?

Qual time ganhou mais jogos em casa na década de 2000?

In [4]:

games2000_win_home = pd.read_sql(
    """SELECT resultado.team_abbreviation_home, resultado.team_name_home, max(resultado.TOTAL_WIN_HOME) as total_win_home
        FROM
        (SELECT team_abbreviation_home, team_name_home, COUNT(*) AS TOTAL_WIN_HOME
      FROM GAME 
       WHERE GAME.game_date BETWEEN "2000-01-01" AND "2009-12-31"
        AND GAME.wl_home = "W"
 GROUP BY GAME.team_abbreviation_home, GAME.team_id_home) as resultado; """, conn)


print(games2000_win_home)

  team_abbreviation_home     team_name_home  total_win_home
0                    SAS  San Antonio Spurs             328


**San Antonio Spurs - 328 home win**

**San Antonio Spurs - 328 vitórias em casa**

# **Decada 2000 - Question 6**


Which team won the most away games in the 2000s?

Qual time ganhou mais jogos fora de casa na década de 2000?

In [5]:
games2000_win_away = pd.read_sql(
    """SELECT resultado.team_abbreviation_away, resultado.team_name_away, max(resultado.TOTAL_WIN_AWAY) as total_win_away
        FROM
        (SELECT team_abbreviation_away, team_name_away, COUNT(*) AS TOTAL_WIN_AWAY
      FROM GAME 
       WHERE GAME.game_date BETWEEN "2000-01-01" AND "2009-12-31"
        AND GAME.wl_away = "W"
 GROUP BY GAME.team_abbreviation_away, GAME.team_id_away) as resultado; """, conn)

print(games2000_win_away)


  team_abbreviation_away     team_name_away  total_win_away
0                    SAS  San Antonio Spurs             245


**San Antonio Spurs - 245 away win**

**San Antonio Spurs - 245 vitórias fora de casa**

# **Decada 2010 - Question 7**

Which team won the most home games in the 2010s?

Qual time ganhou mais jogos em casa na década de 2010?

In [6]:
games2010_win_home = pd.read_sql(
    """SELECT resultado.team_abbreviation_home, resultado.team_name_home, max(resultado.TOTAL_WIN_HOME) as total_win_home
        FROM
        (SELECT team_abbreviation_home, team_name_home, COUNT(*) AS TOTAL_WIN_HOME
      FROM GAME 
       WHERE GAME.game_date BETWEEN "2010-01-01" AND "2019-12-31"
        AND GAME.wl_home = "W"
 GROUP BY GAME.team_abbreviation_home, GAME.team_id_home) as resultado; """, conn)


print(games2010_win_home)

  team_abbreviation_home     team_name_home  total_win_home
0                    SAS  San Antonio Spurs             291


**San Antonio Spurs - 291 home win**

**San Antonio Spurs - 291 vitórias em casa**

# **Decada 2010 - Question 8**


Which team won the most away games in the 2010s?

Qual time ganhou mais jogos fora de casa na década de 2010?

In [7]:
games2010_win_away = pd.read_sql(
    """SELECT resultado.team_abbreviation_away, resultado.team_name_away, max(resultado.TOTAL_WIN_AWAY) as total_win_away
        FROM
        (SELECT team_abbreviation_away, team_name_away, COUNT(*) AS TOTAL_WIN_AWAY
      FROM GAME 
       WHERE GAME.game_date BETWEEN "2010-01-01" AND "2019-12-31"
        AND GAME.wl_away = "W"
 GROUP BY GAME.team_abbreviation_away, GAME.team_id_away) as resultado; """, conn)

print(games2010_win_away)


  team_abbreviation_away     team_name_away  total_win_away
0                    SAS  San Antonio Spurs             215


**San Antonio Spurs - 215 away win**

**San Antonio Spurs - 215 vitórias fora de casa**